### PR TITLE
Add value tracking service

### PR DESCRIPTION
**Pull request by ryzngard**
Add the initial implementation of the value tracking service into the feature branch. The design here is to showcase initial expansion based on entry point from a user. The service returns a set of references that may be further expanded if needed, but is currently not implemented to do so. Does not include UI

### CODE REVIEW

File: src/EditorFeatures/Core/ValueTracking/ValueTrackedItem.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.ValueTracking
+{
+    internal class ValueTrackedItem
```
Comment (jmarolf) [inline src/EditorFeatures/Core/ValueTracking/ValueTrackedItem.cs:7]: Could this be a record?

File: src/EditorFeatures/Core/ValueTracking/ValueTrackingProgressCollector.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.ValueTracking
+{
+    internal class ValueTrackingProgressCollector
```
Comment (jmarolf) [inline src/EditorFeatures/Core/ValueTracking/ValueTrackingProgressCollector.cs:11]: Perhaps this should inherit from `IProgress<T>`?

**Review by jmarolf — APPROVED** [review]
:shipit:

File: src/EditorFeatures/Core/ValueTracking/ValueTrackedItem.cs (hunk 1 above)
Comment (ryzngard) [reply src/EditorFeatures/Core/ValueTracking/ValueTrackedItem.cs:7]: Sure, why not :)

Comment (ryzngard) [reply src/EditorFeatures/Core/ValueTracking/ValueTrackedItem.cs:7]: Actually, can we use records in Roslyn yet? 

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0518	Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported	Microsoft.CodeAnalysis.EditorFeatures (netcoreapp3.1)
```

**Pull request merged**